alicat/serial.py

```diff
@@ -132,21 +132,175 @@ def set_gas(self, gas, retries=2):
+            self._write_and_read(command, retries)
+        else:
+            if gas not in self.gases:
+                raise ValueError("{} not supported!".format(gas))
+            command = '{addr}$${gas}\r'.format(addr=self.address, gas=self.gases.index(gas))
+            self._write_and_read(command, retries)
+
+        read_reg46 = self._write_and_read('{addr}$$R46\r'.format(addr=self.address), retries)
+        reg46 = int(read_reg46.split()[-1])
+        bits = [32768, 16384, 8192, 4096, 2048, 1024, 512]
+
+        for u in range(0, 7):
+            reg46 = reg46 - bits[u]
+            if reg46 < 0:
+                reg46 = reg46 + bits[u]
+            elif reg46 < 255:
+                break
+
+        if num is True:
+            if gas != reg46:
+                raise IOError("Cannot set gas, gas set to Air.")
+        else:
+            if self.gases.index(gas) != reg46:
+                raise IOError("Cannot set gas.")
+
+    def create_mix(self, mix_no, name, gas1, percent1, gas2, percent2, gas3=None, percent3=None, gas4=None, percent4=None, gas5=None, percent5=None, retries=2):
+        """Create a gas mix.
+
+        Gas mixes are made using COMPOSER software located from the front panel and over serial.
+        COMPOSER mixes can only be made on Alicat devices with firmware 5v or greater.
+
+        Args:
+            mix_no: The mix number. Gas mixes are stored in slots 236-255. A mix number of 0 will create a mix in
+            the earliest available spot.
+            name: A name for the gas that will appear on the front panel. Names greater than six letters will be
+            cut off.
+            gas#: Name of the gas, as a string. Supported gas types are:
+                'Air', 'Ar', 'CH4', 'CO', 'CO2', 'C2H6', 'H2', 'He', 'N2',
+                'N2O', 'Ne', 'O2', 'C3H8', 'n-C4H10', 'C2H2', 'C2H4',
+                'i-C2H10', 'Kr', 'Xe', 'SF6', 'C-25', 'C-10', 'C-8', 'C-2',
+                'C-75', 'A-75', 'A-25', 'A1025', 'Star29', 'P-5'
+            percent#: The percentage of the mix for the corresponding gas."""
+
+        self._test_controller_open()
+
+        read = '{addr}VE\r'.format(addr=self.address)
+        firmware = self._write_and_read(read, retries)
+        if "2v" in firmware or "3v" in firmware or "4v" in firmware or "GP" in firmware:
+            raise IOError("This unit does not support COMPOSER gas mixes.")
+
+        if mix_no < 236 or mix_no > 255:
+            raise ValueError("Mix number must be between 236-225!")
+
+        total_percent = 0
+        mix_list = [percent1, percent2, percent3, percent4, percent5]
+        for i in range(0, 5):
+            if mix_list[i] is not None:
+                total_percent += mix_list[i]
+        if total_percent != 100:
+            raise ValueError("Percentages of gas mix must add to 100%!")
+
+        if gas1 is not None and gas1 not in self.gases:
+            raise ValueError("{} not supported!".format(gas1))
```

Comment:
With a dictionary inputted into the function, this could be something like:

```python
if any(gas not in self.gases for gas in gases):
    raise ValueError("{} not supported!".format(gas))
```

(no Nones to handle either)